File: Algorithms/Easy/1684_CountTheNumberOfConsistentStrings/solution.py
from typing import List


class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        not_match=0
        for word in words:
            for w in word:
                if w not in allowed:
                    not_match+=1
                    break
        ans=len(words)-not_match
        return ans

File: Algorithms/Easy/1684_CountTheNumberOfConsistentStrings/Solution.cpp
class Solution {
public:
    int markCharacters(string &word) {
        int marked = 0;
        
        for (char ch : word) {
            int position = ch - 'a';
            marked |= 1 << position;
        }
        
        return marked;
    }
        
    int countConsistentStrings(string allowed, vector<string>& words) {
        int markedAllowed = markCharacters(allowed);
        int answer = 0;
        
        for (string word : words) {
            int markedWord = markCharacters(word);
            
            if ((markedWord & markedAllowed) == markedWord)
                answer++;
        }
        
        return answer;
    }
};